import_flow_events.py

```diff
@@ -100,12 +101,15 @@
 """
 
 Q_CLEAR_DAY_METADATA = """
-    DELETE FROM flow_metadata
-    WHERE begin_time::DATE = '{day}'::DATE;
+    DELETE m
+    FROM flow_metadata AS m
+    LEFT JOIN flow_events AS e
+      ON m.flow_id = e.flow_id
+    WHERE e.flow_id IS NULL;
```

Comment:
Ah OK, I think I see - we delete the event data first, and then we delete any metadata for flows that not longer have any events.  I think this will work well for the case where you're re-importing the later half of a flow, but I'm not sure about when you're importing the earlier half (i.e. you're deleting and re-importing the flow.begin event, but not re-importing the events that complete it).  But it's not obvious what should happen in that case anyway.